app/models/concerns/project_media_cached_fields.rb

```diff
@@ -644,7 +644,8 @@ def recalculate_report_status
     end
 
     def recalculate_tags_as_sentence
-      self.get_annotations('tag').map(&:load).map(&:tag_text).uniq.join(', ')
+      tag_text_ids = Tag.where(annotation_type: 'tag', annotated_type: 'ProjectMedia', annotated_id: self.id).collect{|t| t.data['tag']}
```

Comment:
Question: Would it be worth it to extract this to a method like `get_tag_annotations`, or is this silly? I ask because I like how it was written before, but it is still pretty clear what we are doing now.

Reply:
No need for this, as get_annotations('tag') performs the same logic; however, I want to avoid the N+1 query.